### ISSUE TITLE
[ReadSupport] Add support to read from nodes with provided labels

### ISSUE THREAD

**Issue by conker84** [neo4j-contrib/neo4j-spark-connector#141]
We want read nodes as tables in the following way:

```scala
val simpleDf = sparkSession
   .read
   .format("neo4j")
   .option("url", "neo4j://localhost:7687")
   .option("node", "Person")
   .load()
```

Example:
For the given set of nodes:
```cypher
CREATE (p:Person:Customer{name: 'Andrea', surname: 'Santurbano'})
CREATE (p:Person{name: 'Davide', surname: 'Fantuzzi'})
```

The output of the Dataframe will be the following:

`<id>` | `<labels>` | `name` | `surname`
-- | -- | -- | --
0 | [Person,Customer] | Andrea | Santurbano
1| [Person] | Davide | Fantuzzi

N.b. this feature will support push-down filtering in order to optimize the data moved from Neo4j to Spark

**Comment by moxious** [comment]
recommend you change the option name to "label" not node because this is a more accurate description of what the option actually is.   Also it seems it's maybe either a string or an array of strings.  It should work with an array too.

Just for completeness, output dataframe should also include the neo4j internal id.  We need a standard fieldname for representing the neo4j internal ID.  (I'd vote for `_id`)

**Comment by conker84** [comment]
@moxious if is not a problem we want to use `<id>` because is like how the Neo4j browser shows the field and because it's less common compared to `_id`, someone could use that name for a field in the relational model for instance.

**Comment by conker84** [comment]
closed via 83d4d28
